aws/table_aws_route53_vpc_association_authorization.go

```diff
@@ -0,0 +1,144 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/route53"
+
+	"github.com/turbot/steampipe-plugin-sdk/v5/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin/transform"
+)
+
+func tableAwsRoute53VPCAssociationAuthorization(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_route53_vpc_association_authorization",
+		Description: "AWS Route53 VPC Association Authorization",
+		List: &plugin.ListConfig{
+			Hydrate: listVPCAssociationAuthorization,
+			Tags:    map[string]string{"service": "route53", "action": "ListVPCAssociationAuthorizations"},
+			KeyColumns: plugin.KeyColumnSlice{
+				{
+					Name:    "hosted_zone_id",
+					Require: plugin.Required,
+				},
+			},
+		},
+		Columns: awsGlobalRegionColumns([]*plugin.Column{
+			{
+				Name:        "hosted_zone_id",
+				Description: "The ID of the hosted zone for which you want a list of VPCs that can be associated with the hosted zone.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "vpc_id",
+				Description: "(Private hosted zones only) The ID of an Amazon VPC.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("VPCId"),
+			},
+			{
+				Name:        "vpc_region",
+				Description: "(Private hosted zones only) The region that an Amazon VPC was created in.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("VPCRegion"),
+			},
+
+			// Steampipe standard columns
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("HostedZoneId", "VPCId"),
+			},
+			// no tags for this resource as per aws docs
+			{
+				Name:        "akas",
+				Description: resourceInterfaceDescription("akas"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.From(authorizationAkas).Transform(transform.ToString),
+			},
+		}),
+	}
+}
+
+type VPCAssociationAuthorizationResult struct {
+	HostedZoneId *string
+	VPCId        *string
+	VPCRegion    *string
+}
+
+func authorizationAkas(ctx context.Context, d *transform.TransformData) (interface{}, error) {
+	result := d.HydrateItem.(*VPCAssociationAuthorizationResult)
+
+	// same format as terraform builtin IDs "HOSTED_ZONE_ID:VPC_ID"
+	// https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/route53_vpc_association_authorization#import
+	akas := fmt.Sprintf("%s:%s", *result.HostedZoneId, *result.VPCId)
+	return akas, nil
+}
+
+// // LIST FUNCTION
+func listVPCAssociationAuthorization(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+
```

Comment:
Could you please update the list function as follows?

- Use the function [listHostedZones](https://github.com/turbot/steampipe-plugin-aws/blob/f4578bfe1aef0c4717a0cbfc71cbbe5421e05aee/aws/table_aws_route53_zone.go#L173) as ParentHydrate for this table.
- Set the column `hosted_zone_id` as the option key quals in the list config.
- Add a check to reduce the number of API calls if the `hosted_zone_id` is provided as a query parameter.

Ref: https://github.com/turbot/steampipe-plugin-aws/blob/f4578bfe1aef0c4717a0cbfc71cbbe5421e05aee/aws/table_aws_route53_record.go#L146-L153